crates/ra_db/src/input.rs

```diff
@@ -165,7 +165,7 @@ impl CrateGraph {
         self.arena.is_empty()
     }
 
-    pub fn iter<'a>(&'a self) -> impl Iterator<Item = CrateId> + 'a {
+    pub fn iter(&self) -> impl Iterator<Item = CrateId> + '_ {
```

Comment:
Am I right to suppose that `'_` lifetime is required here but not in `fn dependencies()`, because the item of an iterator in `dependencies` is a reference of `self` lifetime and it forces the compiler to infer `impl Iterator<...> + lifetimeof(self)` there but not here?...

Reply:
Seems so, although I don't know precise rules here